scripts/directory-to-volume.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+# copy the contents of a directory into a new docker volume
+
+set -e
+
+if [ "$#" -ne 2 ]; then
+    echo "Usage $0 <source> <destination>"
+    exit 1
+fi
+
+SRC="$1"
+DEST="$2"
+
+if [ ! -d $SRC ]; then
+    echo "Error: path $SRC must be a directory"
+    exit 1
+fi
+
+if docker volume inspect "$DEST" > /dev/null 2>&1; then
+    echo "Error: docker volume $DEST already exists"
+    exit 1
+fi
+
+SRC=$(realpath $SRC)
+
+echo -n "Creating docker volume "
+docker volume create "$DEST"
+
+echo "Copying files"
+docker run \
+       --rm \
+       -v "$SRC":/src \
+       -v "$DEST":/dest \
+       alpine:latest \
+       ash -c "cd /src; tar cf - . | (cd /dest && tar xf -)"
+
+echo "Done"
```

Comment:
I've tested this locally and I'm happy with it. This might be overkill, but it occurs to me that this would be quite easy to test via TeamCity. Something like:

```
mkdir test
echo "Hello world" > test/hi.txt
./scripts/directory-to-volume.sh ~/Desktop/test/ test_volume
result=$(docker run -it --rm -v test_volume:/test alpine ash -c "cat /test/hi.txt")
rm -r test
docker volume rm test_volume
if [ $result -ne "Hello world" ]; then
    echo "Test failed"
    exit -1
else
    echo "Test passed"
    exit 0
fi
```

But maybe the test is just as complicated as the script, and thus doesn't add much confidence?

Reply:
My preference would probably be to (eventually) collect up the utility scripts we use that are not tied to a particular repo and test them in isolation - there's a similar script for duplicating a volume in montagu too (https://github.com/vimc/montagu-registry/blob/master/scripts/docker_clone_volume.sh) - the script itself is not used routinely but just in this one migration

Reply:
:+1: